src/sql/workbench/browser/modelComponents/button.component.ts

```diff
@@ -24,33 +24,44 @@ import { convertSize } from 'sql/base/browser/dom';
 @Component({
 	selector: 'modelview-button',
 	template: `
+	<p>BT: {{this._buttonType}}</p>
 	<div>
 		<label for={{this.label}}>
 			<div #input style="width: 100%">
 				<input #fileInput *ngIf="this.isFile === true" id={{this.label}} type="file" accept="{{ this.fileType }}" style="display: none">
 			</div>
 		</label>
 	</div>
+
 	`
 })
+
 export default class ButtonComponent extends ComponentWithIconBase implements IComponent, OnDestroy, AfterViewInit {
 	@Input() descriptor: IComponentDescriptor;
 	@Input() modelStore: IModelStore;
 	private _button: Button;
+	public _buttonType = <azdata.ButtonType>'Normal';
 	public fileType: string = '.sql';
 
 	@ViewChild('input', { read: ElementRef }) private _inputContainer: ElementRef;
 	@ViewChild('fileInput', { read: ElementRef }) private _fileInputContainer: ElementRef;
+
 	constructor(
 		@Inject(forwardRef(() => ChangeDetectorRef)) changeRef: ChangeDetectorRef,
 		@Inject(IWorkbenchThemeService) private themeService: IWorkbenchThemeService,
 		@Inject(forwardRef(() => ElementRef)) el: ElementRef
 	) {
 		super(changeRef, el);
+
+		if (this.isFile === true || this.buttonType === 'File') {
```

Comment:
don't do this in the constructor, the properties are not set at this point. you should only check IsFile in the getter of ButtonType and also, you shouldn't need to call the setpropertyfromUI